modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/raft/ZonePartitionRaftListener.java

```diff
@@ -266,6 +270,10 @@ private IgniteBiTuple<Serializable, Boolean> processCrossTableProcessorsCommand(
             long commandTerm,
             @Nullable HybridTimestamp safeTimestamp
     ) {
+        if (tableProcessors.isEmpty()) {
+            return new IgniteBiTuple<>(null, true);
```

Comment:
Shouldn't it be `false` instead? And I think this check is redundant anyway.

Reply:
Fixed as we discussed yesterday: we should check the local in-memory `lastAppliedIndex` and if it lower than the received -- update index & term and then adjust the time.